src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -341,13 +341,13 @@ protected virtual TxReceipt[] ProcessBlock(
         // We need to do a commit here as in _executionRequestsProcessor while executing system transactions
         // we do WorldState.Commit(SystemTransactionReleaseSpec.Instance). In SystemTransactionReleaseSpec
         // Eip158Enabled=false, so we end up persisting empty accounts created while processing withdrawals.
-        _stateProvider.Commit(spec);
+        _stateProvider.Commit(spec, commitRoots: false);
 
         _executionRequestsProcessor.ProcessExecutionRequests(block, _stateProvider, receipts, spec);
 
         ReceiptsTracer.EndBlockTrace();
 
-        _stateProvider.Commit(spec, commitStorageRoots: true);
+        _stateProvider.Commit(spec, commitRoots: true);
```

Comment:
Am I getting it right that after these, right now, when we access the root hash will be wrong until the final `commitRoots: true` is called, right? I don't see it as a problem in any execution path, but maybe it is?

Reply:
Will be currently anyway unless pre-Eip658; doesn't change behaviour in this PR either; if Eip658 not enabled then it will commit the roots at end of tx